### PR TITLE
update shuffleboard port

### PR DESCRIPTION
**Pull request by ariovistus**

### CODE REVIEW

File: wpilib/tests/test_shuffleboard.py [hunk 1]
```diff
@@ -22,8 +30,7 @@ def test_enableActuatorWidgets(wpilib, getTable):
 def test_addPersistent(wpilib, getTable):
     tab = wpilib.shuffleboard.Shuffleboard.getTab("tacos")
     widget = tab.addPersistent("simplevalue", 1)
-    # k, how to kick networktables?
-    # assert widget.getEntry().isPersistent()
+    assert widget.getEntry().isPersistent()
```
Comment (ariovistus) [inline wpilib/tests/test_shuffleboard.py]: need help here, why is this failing?

File: wpilib/tests/test_shuffleboard.py [hunk 2]
```diff
@@ -79,3 +86,125 @@ def test_withWidget(wpilib, getTable):
...
+    assert rc.recordingControlEntry.getBoolean(None) == True
+
+
+def test_RecordingController_setRecordingFileNameFormat(recordingcontroller):
+    rc = recordingcontroller
+    assert rc.recordingFileNameFormatEntry.getString(None) is None
+    rc.setRecordingFileNameFormat("tacos")
+    assert rc.recordingFileNameFormatEntry.getString(None) == "tacos"
+    rc.clearRecordingFileNameFormat()
+    assert rc.recordingFileNameFormatEntry.getString(None) is None
```
Comment (ariovistus) [inline wpilib/tests/test_shuffleboard.py]: need help here, why is this one failing?

File: wpilib/tests/test_shuffleboard.py (hunk 1 above)
Comment (virtuald) [reply wpilib/tests/test_shuffleboard.py]: pynetworktables bug, oops...

Comment (virtuald) [reply wpilib/tests/test_shuffleboard.py]: See https://github.com/robotpy/pynetworktables/pull/76

File: wpilib/tests/test_shuffleboard.py [hunk 3]
```diff
@@ -1,11 +1,19 @@
...
 
 
 @pytest.fixture(scope="function")
 def getTable(networktables):
     return networktables.NetworkTables.getTable
 
 
+@pytest.fixture(scope="function")
+def recordingcontroller(wpilib, networktables):
+    nt_instance = networktables.instance.NetworkTablesInstance.getDefault()
```
Comment (auscompgeek) [inline wpilib/tests/test_shuffleboard.py]: Wait, you realise this is the same as `networktables.NetworkTables`, right?

**Pull request merged**